File: python/meteroid/models/create_coupon_request.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .coupon_discount import CouponDiscount
from .plan_id import PlanId


@dataclasses.dataclass
class CreateCouponRequest(BaseModel):
    code: str

    discount: CouponDiscount

    description: t.Optional[str] = None

    expires_at: t.Optional[datetime] = None

    plan_ids: t.Optional[t.List[PlanId]] = None

    recurring_value: t.Optional[int] = None

    redemption_limit: t.Optional[int] = None

    reusable: t.Optional[bool] = None

File: go/create_coupon_request.go
// this file is @generated
package meteroid

import "time"

type CreateCouponRequest struct {
	Code string `json:"code"`

	Description *string `json:"description,omitempty"`

	Discount CouponDiscount `json:"discount"`

	// RFC 3339 timestamp.
	ExpiresAt *time.Time `json:"expires_at,omitempty"`

	PlanIds []PlanId `json:"plan_ids,omitempty"`

	RecurringValue *int32 `json:"recurring_value,omitempty"`

	RedemptionLimit *int32 `json:"redemption_limit,omitempty"`

	Reusable *bool `json:"reusable,omitempty"`
}
